slobrok/src/vespa/slobrok/server/exchange_manager.h

```diff
@@ -81,41 +71,36 @@ class ExchangeManager
         void addItem(RemoteSlobrok *partner);
         void doneItem(bool denied);
         void expedite();
-        WorkPackage(op_type op,
-                    const char *name, const char *spec,
-                    ExchangeManager &exchanger,
-                    RegRpcSrvCommand  donehandler);
+        WorkPackage(const WorkPackage&) = delete;
+        WorkPackage& operator= (const WorkPackage&) = delete;
+        WorkPackage(op_type op, const std::string & name, const std::string & spec,
+                    ExchangeManager &exchanger, RegRpcSrvCommand  donehandler);
         ~WorkPackage();
     };
 
 public:
+    ExchangeManager(const ExchangeManager &) = delete;
+    ExchangeManager &operator=(const ExchangeManager &) = delete;
     ExchangeManager(SBEnv &env, RpcServerMap &rpcsrvmap);
-    ~ExchangeManager() {}
+    ~ExchangeManager();
 
     SBEnv             &_env;
     RpcServerManager  &_rpcsrvmanager;
     RpcServerMap      &_rpcsrvmap;
 
-    OkState addPartner(const char *name, const char *spec);
-    void removePartner(const char *name);
+    OkState addPartner(const std::string & name, const std::string & spec);
+    void removePartner(const std::string & name);
     std::vector<std::string> getPartnerList();
 
-    void registerFrom(RemoteSlobrok *partner);
-    void reregisterTo(RemoteSlobrok *partner);
+    void forwardRemove(const std::string & name, const std::string & spec);
 
-    void forwardRemove(const char *name, const char *spec);
-
-    void wantAdd(const char *name, const char *spec, RegRpcSrvCommand rdc);
-    void doAdd(const char *name, const char *spec, RegRpcSrvCommand rdc);
-
-    RemoteSlobrok *lookupPartner(const char *name) const {
-        return _partners[name];
-    }
+    void wantAdd(const std::string & name, const std::string & spec, RegRpcSrvCommand rdc);
```

Comment:
Command -> Data

Reply:
Not possible as Data is visible outside of cmd.cpp